File: Parking Lot/Commands/CreateParkingLotCommandExecutor.cs
using System.Collections.Generic;
using Model;
using Model.Parking.Strategy;
using Service;

namespace Commands
{

    /**
     * Executor to handle command of creating the initial parking lot.
     */
    public class CreateParkingLotCommandExecutor : CommandExecutor
    {
        public static String COMMAND_NAME = "create_parking_lot";

        public CreateParkingLotCommandExecutor(
             ParkingLotService parkingLotService, OutputPrinter outputPrinter) : base(parkingLotService, outputPrinter)
        {

        }

        /**
         * {@inheritDoc}
         */
        public override bool Validate(Command command)
        {
            List<string> paramList = command.Params;
            if (paramList.Count != 1) {
                return false;
            }
            return int.TryParse(paramList[0], out int value);

        }

        /**
         * {@inheritDoc}
         */
        public override void Execute(Command command)
        {
            int.TryParse(command.Params[0], out int parkingLotCapacity);
            ParkingLot parkingLot = new ParkingLot(parkingLotCapacity);
            parkingLotService.createParkingLot(parkingLot, new NaturalOrderingParkingStrategy());
            outputPrinter.PrintWithNewLine(
                "Created a parking lot with " + parkingLot.Capacity + " slots");
        }
    }
}

File: Parking Lot/Parking Lot Java/src/main/java/org/example/command/CreateParkingLotCommandExecutor.java
package org.example.command;

import lombok.AllArgsConstructor;
import org.example.ParkingLotService;
import org.example.exceptions.ParkingLotExistsException;
import org.example.models.parking_stratergy.NaturalOrderParkingStrategy;

import java.util.List;

@AllArgsConstructor
public class CreateParkingLotCommandExecutor implements CommandExecutor {

    final ParkingLotService parkingLotService;

    @Override
    public void execute(List<String> params) {

        try {
            this.parkingLotService.createParkingLot(Integer.valueOf(params.getFirst()),
                    new NaturalOrderParkingStrategy());
            System.out.println("Created Parking Lot with capacity: " + params.getFirst());
        }
        catch (ParkingLotExistsException exception) {
            System.out.println("Parking lot already Exists!");
        }
    }

    @Override
    public boolean validate(List<String> params) {
        return params.size() == 1;
    }
}
